src/Codeception/Module/Symfony2.php

```diff
@@ -174,6 +178,46 @@ protected function getKernelClass()
     }
 
     /**
+     * Get router from container.
+     *
+     * @return object
+     */
+    private function getNonCachedRouter()
+    {
+        if (!$this->kernel->getContainer()->has('router')) {
+            $this->fail('Router not found.');
+        }
+
+        return $this->kernel->getContainer()->get('router');
+    }
+
+    /**
+     * Helper method to get router.
+     * Router can be cached or not by configuration.
+     *
+     * @return object
+     */
+    protected function getRouter()
+    {
+        if ($this->config['cache_router'] === true) {
+            if (null === self::$router) {
+                self::$router = $this->getNonCachedRouter();
+            }
+            return self::$router;
+        }
+
+        return $this->getNonCachedRouter();
+    }
+
+    /**
+     * Invalidate previously cached router.
+     */
+    public function invalidateCachedRouter()
```

Comment:
Yes, but if you have enabled it, this is the only way to manually invalidate the cached routes.
I am not sure if this method could be useful or not, it would depend on specific use case.

So I think that is safe to give to the user the option to force cache invalidation, we don't know all scenarios that our users are executing, may be some users have dynamic routes and need a way to invalidate cache only at certain tests... This way it is safe to enable cached routes and invalidate cache on specific tests.

If nobody will use this feature, then it is safe to delete the method.